server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -833,14 +834,26 @@ public Symbol visitArrayLikePredicate(ArrayLikePredicate node, ExpressionAnalysi
 
         @Override
         protected Symbol visitLikePredicate(LikePredicate node, ExpressionAnalysisContext context) {
-            if (node.getEscape() != null) {
-                throw new UnsupportedOperationException("ESCAPE is not supported.");
-            }
             Symbol expression = node.getValue().accept(this, context);
             Symbol pattern = node.getPattern().accept(this, context);
+            Symbol escape;
+            String escapeValue;
+            if (node.getEscape() != null) {
+                escape = node.getEscape().accept(this, context);
+                // Aligned with PG: an empty ESCAPE falls back to default backslash.
+                assert escape instanceof Literal : "ESCAPE must be provided as a literal.";
```

Comment:
In the grammar, `escape` is a valueExpression, so it can also be a column ident, shouldn't we restrict it in the grammar, and have it simply a `parameterOrLiteral`?

Reply:
Also if it is a parameter (`?`) bound later, would it work, can you try an integration test for that?

Reply:
https://github.com/crate/crate/pull/15166/commits/748a7966522f6dea965c9959a04494f9c8e39aec

I will remove integration test later, ok?